keras_cv/layers/preprocessing/random_width.py

```diff
@@ -0,0 +1,133 @@
+      interpolation: String, the interpolation method. Defaults to `bilinear`.
+        Supports `"bilinear"`, `"nearest"`, `"bicubic"`, `"area"`, `"lanczos3"`,
+        `"lanczos5"`, `"gaussian"`, `"mitchellcubic"`.
+      seed: Integer. Used to create a random seed.
+
+    Input shape:
+      3D (unbatched) or 4D (batched) tensor with shape:
+      `(..., height, width, channels)`, in `"channels_last"` format.
+
+    Output shape:
+      3D (unbatched) or 4D (batched) tensor with shape:
+      `(..., height, random_width, channels)`.
+    """
+
+    def __init__(self, factor, interpolation=tf.image.ResizeMethod.BILINEAR, seed=None, **kwargs):
+        super().__init__(seed=seed, force_generator=True, **kwargs)
+        self.factor = factor
+        if isinstance(factor, (tuple, list)):
+            self.width_lower = factor[0]
+            self.width_upper = factor[1]
+        else:
+            self.width_lower = -factor
+            self.width_upper = factor
+        if self.width_upper < self.width_lower:
+            raise ValueError(
+                "`factor` cannot have upper bound less than "
+                "lower bound, got {}".format(factor)
+            )
+        if self.width_lower < -1.0 or self.width_upper < -1.0:
+            raise ValueError(
+                "`factor` must have values larger than -1, " "got {}".format(factor)
+            )
+        self.interpolation = interpolation
+        self._interpolation_method = preprocessing.get_interpolation(
+            interpolation=interpolation
+        )
+        self.seed = seed
+        self.auto_vectorize = False
+
+    def augment_label(self, label, transformation=None, **kwargs):
+        return label
+
+    def _batch_augment(self, inputs):
+        images = self.augment_image(
+            inputs[IMAGES],
+            transformation=self.get_random_transformation(image=inputs[IMAGES]),
+        )
+        result = {IMAGES: images}
+        return result
+
+    def augment_image(self, image, transformation, bounding_boxes=None, label=None):
+        # The batch dimension of the input=image is not modified. The output
+        # should be accurate for both unbatched and batched input
+        inputs = preprocessing.ensure_tensor(image)
+        inputs_shape = tf.shape(inputs)
+        img_height_dim = inputs_shape[HEIGHT_AXIS]
+        adjusted_width = transformation["width"]
+        adjusted_size = tf.stack([img_height_dim, adjusted_width])
+        output = tf.image.resize(
+            images=inputs, size=adjusted_size, method=self._interpolation_method
+        )
+        # tf.resize will output float32 in many cases regardless of input type.
+        output = tf.cast(output, self.compute_dtype)
+        output_shape = inputs.shape.as_list()
+        output_shape[WIDTH_AXIS] = None
+        output.set_shape(output_shape)
+        return output
+
+    def get_random_transformation(
+        self,
+        image=None,
+        label=None,
```

Comment:
I don't think this actually works because RandomWidth() operates on a batch-wise basis.  This looks like it will operate on a sample-wise basis.

Perhaps we can break API compat and return on a sample-wise basis, but we'd need to ask @fchollet.

Reply:
ah wait, nevermind you're only calling it once in batch_augment.  apologies

Reply:
So, is this fine as is?